lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -304,21 +304,24 @@ public function match($token)
         $lookaheadType = $this->lexer->lookahead['type'];
 
         // Short-circuit on first condition, usually types match
-        if ($lookaheadType !== $token) {
-            // If parameter is not identifier (1-99) must be exact match
-            if ($token < Lexer::T_IDENTIFIER) {
-                $this->syntaxError($this->lexer->getLiteral($token));
-            }
+        if ($lookaheadType === $token) {
```

Comment:
I can drop this commit if you prefer, it was just to make things a bit more sane.

Reply:
Works as-is

Reply:
It works indeed, it's just not so good as it could be... I can send another PR to make you happier 😂